modules/ide/xsk-ide-xsaccess/src/main/resources/META-INF/dirigible/xsk-ide-xsaccess/editor/editor.js

```diff
@@ -0,0 +1,169 @@
+        return getResource('/services/v4/ide/workspaces' + xsaccessfile);
+      }
+    }
+
+    function getViewParameters() {
+      if (window.frameElement.hasAttribute("data-parameters")) {
+        let readParams = JSON.parse(window.frameElement.getAttribute("data-parameters"));
+        $scope.file = readParams["file"];
+      } else {
+        let scanSearchParams = new URLSearchParams(window.location.search);
+        $scope.file = scanSearchParams.get('file');
+      }
+    }
+
+    $scope.initJSONcontent = function () {
+      if (!('allowMethods' in $scope.data.cors))
+        $scope.data.cors = {
+          ...$scope.data.cors,
+          allowMethods: $scope.allAllowedMethods,
+          allowOrigin: ["*", "*.sap.com"],
+          maxAge: 3600,
+          allowHeaders: [''],
+          exposeHeaders: ['']
+        };
+      if ($scope.data.headers.customHeaders && $scope.data.headers.customHeaders.length)
+        $scope.data.headers.customHeaders = $scope.data.headers.customHeaders.map((x) => {
+          return {
+            ...x,
+            thevalue: x.value.substr(0, 10) == 'ALLOW-FROM' ? 'ALLOW-FROM' : x.value,
+            url: x.value.substr(0, 10) == 'ALLOW-FROM' ? x.value.substr(11) : ''
+          }
+        })
+    }
+
+    function load() {
+      getViewParameters();
+      contents = loadContents($scope.file);
+      $scope.data = JSON.parse(contents);
+      $scope.anonymous_connection_allowed = $scope.data.anonymous_connection ? true : false;
+      $scope.allAllowedMethods = ["GET", "POST", "HEAD", "OPTIONS"];
+      $scope.initJSONcontent();
+    }
+
+    load();
+
+    function updateObjectNestedProp(obj, value, propPath) {
+      const [head, ...rest] = propPath.split('.');
+      if (!rest.length) {
+        obj[head] = value;
+      } else {
+        updateObjectNestedProp(obj[head], value, rest.join('.'));
+      }
+    }
+
+    function getNestedValue(obj, path) {
+      return path.split('.').reduce(function (prev, curr) {
+        return prev[curr];
+      }, obj || this);
+    }
+
+    $scope.deleteInList = function (path, index) {
+      let newKeyValue = getNestedValue($scope.data, path).filter((_val, num) => num != index);
+      if (!newKeyValue.length) newKeyValue.push('');
+      updateObjectNestedProp(
+        $scope.data,
+        newKeyValue,
+        path);
+    }
+
+    $scope.addInList = function (path, value) {
+      let newKeyValue = getNestedValue($scope.data, path);
+      newKeyValue.push(value);
+      updateObjectNestedProp($scope.data,
+        newKeyValue,
+        path);
+    }
+
+    $scope.trigCorsMethod = function (method) {
+      if (!$scope.data.cors.allowMethods.includes(method))
+        $scope.data.cors.allowMethods.push(method);
+      else
+        $scope.data.cors.allowMethods = $scope.data.cors.allowMethods.filter((x) => x != method);
+    }
+
+    function saveContents(contentToSave) {
+      if ($scope.file) {
+        console.log('Save of XSACCESS called...');
```

Comment:
Is this log needed?

Reply:
This is the way we do it in the other editors Guess it may be helpful in case contents is not saved due to a backend failure.

Reply:
Wouldn't the backend request's payload and response be visible in Chrome Dev Tools? Logging in the console should not be used if debugging could be done in another way. Remove this log unless there's a specific reason it should be there